File: dynamic_listings/lua/default/optimization/dirty_bit.lua
Player = {}

function Player:new(o)
    o = o or {
        speed = {0, 0},
        needs_update = false
    }
    setmetatable(o, self)
    self.__index = self
    return o
end

-- ...
function Player:input()
    -- ...
    if self.right_key.is_Pressed then
        self.speed.x = self.speed.x + 1  -- Move right
        self.needs_update = true
        -- ...
    end
    if self.up_key.is_Pressed then
        self.speed.y = self.speed.y -100  -- Move up (jump)
        self.needs_update = true
    end
    -- ...
end

function Player:update(dt)
    if self.needs_update then
        -- Do Update instructions
        -- ...
    end
end

File: dynamic_listings/python/default/optimization/dirty_bit.py
class Player:
    # We assume the presence of a "Vector2" class
    speed: Vector2 = Vector2(0, 0)
    needs_update: bool = False
    # ...

    def get_input(self) -> None:
        # ...
        if right_key.is_pressed:
            speed = speed + Vector2(1, 0)  # Move right
            self.needs_update = True
            # ...
        if up_key.is_pressed:
            speed = speed + Vector2(0, -100)  # Move up (jump)
            self.needs_update = True
        # ...

    def update(self, dt: float) -> None:
        if self.needs_update:
            # Do Update instructions
            # ...
